src/modules/api/user/user.controller.ts

```diff
@@ -1,7 +1,35 @@
-import { Controller } from '@nestjs/common';
+import { Controller, Get, Param } from '@nestjs/common';
 import { UserService } from './user.service';
+import { IResult } from 'src/common/interfaces/response';
 
-@Controller('users/:id')
+@Controller('users')
 export class UserController {
   constructor(private userService: UserService) {}
+
+  @Get(':id')
+  async getUserById(@Param() params) {
+    try {
+      const user = await this.userService.getUserById(params.id);
```

Comment:
You can just get Pramas['id'] instead of passing down the whole object.

Reply:
Changed to @Param('id') id

Reply:
Did you make the necessary changes?